src/web/mjs/connectors/sakuramanga.mjs

```diff
@@ -0,0 +1,82 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class SakuraManga extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'sakuramanga';
+        super.label = 'Sakura Manga';
+        this.tags = ['manga', 'japanese'];
+        this.url = 'https://sakuramanga.net';
+
+        this.list = '/japanese-manga-list';
+    }
+
+    async _getMangaFromURI(uri) {
+        try {
+            let request = new Request(uri, this.requestOptions);
+            let data = await this.fetchDOM(request, 'head title');
+            let id = uri.pathname;
+            let title = data[0].text.split(' | ')[0].trim();
+            return new Manga(this, id, title);
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async _getMangaList(callback) {
+        try {
+            let request = new Request(this.url + '/seri-listesi?type=text', this.requestOptions);
+            let data = await this.fetchDOM(request, 'ul[id*="menu-menu-sidebar-"] li[id*="menu-item-"] a');
+            let mangaList = data.map(element => {
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                    title: element.text.trim()
+                };
+            });
+            callback(null, mangaList);
+        } catch (error) {
+            console.error(error, this);
+            callback(error, undefined);
+        }
+    }
+
+    async _getChapterList(manga, callback) {
+        try {
+            let request = new Request(this.url + manga.id, this.requestOptions);
+            let data = await this.fetchDOM(request, 'article[id*="post-"] header div a');
+            let chapterList = data.map(element => {
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                    title: element.text.trim(),
+                    language: ''
+                };
+            });
+            callback(null, chapterList);
+        } catch (error) {
+            console.error(error, manga);
+            callback(error, undefined);
+        }
+    }
+
+    async _getPageList(manga, chapter, callback) {
+        try {
+            let request = new Request(this.url + chapter.id, this.requestOptions);
+            let data = await this.fetchDOM(request, 'div.entry-content noscript');
```

Comment:
Is it possible to get the image links directly?
e.g.
```javascript
let data = await this.fetchDOM(request, 'div.entry-content source.lazy');
let pageList = data.map(element => this.getAbsolutePath(element.dataset.src, request.url));
callback(null, pageList);
```

Reply:
Not to my knowledge, ive tried the original way you explained in the tutorial but kept getting full page as respond and only a lazy_load.gif
seems to be a wordpress. you may know the answer to that
so i brew my own solution

Reply:
You may try the above code if there may appear any problems in the future.
Current implementation is non-blocking for the pull request.